### ISSUE TITLE
Keep up to date AdHoc provisioning profile, OR how exactly signing works?

### ISSUE THREAD

**Issue by maximkhatskevich** [fastlane/fastlane#1069]
Great thing, but not sure I understand how signing works. Does it overrides/ignores somehow settings from project info.plist and project build/code_signing settings and force apply that settings from Appfile during deployment process (when execute a lane)? Fastlane seems to be duplicating and overriding many things that are presented in Xcode project file and info.plist, I would like to understand better how it works.

Ideally any parameter should be set only once and then used everywhere.

Here is an example.

The case: "keep up to date AdHoc provisioning profile".

Desired workflow (by "set" I mean it should be set explicitly in a configuration file, by "get" I mean it should be retrieved automatically by Fastlane or one of the related tools without asking explicitly from developer):

- set desired WORKSPACE;
- set desired PROJECT;
- set desired SCHEME;
- set desired ACTION from SCHEME ('Run' or 'Archive', other to be supported if needed);
- set desired DEVICE list that need to be included in the provisioning PROFILE;
- set desired code signing IDENTITY (certificate) list that need to be included in the PROFILE in case of re-creation;

- get TARGET from SCHEME;
- get CONFIGURATION from ACTION;
- get provisioning PROFILE from TARGET for given CONFIGURATION;
- get development TEAM from TARGET;
- get list of availalbe accounts (APPLE_ID) from Xcode?, select one automatically (with confirmation?) based on the TEAM;

- for each DEVICE check, if it is registered under the TEAM;
- for each DEVICE check, if it is included in the PROFILE, proceed to next step if find a device that is not included;
- if at least one DEVICE is not included in the PROFILE, then (show summary and ask for confirmation before doing):
	- if IDENTITY list is not set explicitly, then get signing IDENTITY (based on the TEAM, APPLE_ID and PROFILE somehow);
	- re-generate provisioning PROFILE on dev. portal (remotely),
	- download and install it in Xcode (Keychain?).

Comments:
	After these steps are completed, developer would need manually update signing settings of the target and perform Git/GitFlow actions, if needed. This approach suppose to NOT download/store provisioning profile locally (to the project folder). Code signing identity supposed to be "Automatic".

====

Any ideas how I can achieve the functionality described above?

**Comment by vpolouchkine** [comment]
It looks like there hasn’t been any recent activity on this issue in some time, so I’m closing it. If this is still a problem, feel free to re-open a new issue and I’d be happy to help :+1:
